scheduler/server/task_runner_test.go

```diff
@@ -201,6 +201,37 @@ func Test_runTaskAndLog_MarkFailedTaskAsFinished(t *testing.T) {
 	}
 }
 
+// Test_cannotGetStatusFromWorkerReturnsFlakyResult verify that the result has an error and
+// state is not runner.Failed (this triggers stateful_scheduler to mark the worker as flaky)
+func Test_cannotGetStatusFromWorkerReturnsFlakyResult(t *testing.T) {
+	log.Debug("Test_cannotGetStatusFromWorkerReturnsFlakyResult")
+	task := domain.GenTask()
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	// setup mock worker that returns an error.
+	chaos := runners.NewChaosRunner(nil)
+	chaos.SetError(fmt.Errorf("connection error"))
+	chaos.SetRunStatus(runner.RunStatus{State: runner.UNKNOWN})
+
+	// set up a mock saga log that verifies task is started and completed with a task with Unknown
+	sagaLogMock := saga.NewMockSagaLog(mockCtrl)
+	sagaLogMock.EXPECT().StartSaga("job1", nil)
+	msgMatcher := TaskMessageMatcher{Type: &sagaStartTask, JobId: "job1", TaskId: "task1", Data: gomock.Any()}
+	sagaLogMock.EXPECT().LogMessage(msgMatcher)
+	msgMatcher = TaskMessageMatcher{Type: &sagaEndTask, JobId: "job1", TaskId: "task1", Data: gomock.Any()}
+	sagaLogMock.EXPECT().LogMessage(msgMatcher)
+
+	sagaCoord := saga.MakeSagaCoordinator(sagaLogMock)
+	s, _ := sagaCoord.MakeSaga("job1", nil)
+
+	err := get_testTaskRunner(s, chaos, "job1", "task1", task, true, stats.NilStatsReceiver()).run()
+
+	assert.NotNil(t, err.(*taskError).runnerErr)
+	assert.Equal(t, err.(*taskError).st.State, runner.UNKNOWN)
```

Comment:
returning UNKNOWN will trigger setting flaky flag to true in the task callback in stateful_scheduler.go